addons/nuxeo-csv/nuxeo-csv-core/src/test/java/org/nuxeo/ecm/csv/core/operation/TestImportOperation.java

```diff
@@ -131,15 +136,67 @@ public void testImportOperation() throws OperationException, InterruptedExceptio
         chain = new OperationChain("test-chain");
         chain.add(CSVImportResultOperation.ID);
 
-        ctx = new OperationContext(session);
-        ctx.setInput(importId);
-
-        CSVImportResult result = (CSVImportResult) service.run(ctx, chain);
+        CSVImportResult result;
+        try (OperationContext ctx = new OperationContext(session)) {
+            ctx.setInput(importId);
+            result = (CSVImportResult) service.run(ctx, chain);
+        }
 
         assertNotNull(result);
         assertEquals(0, result.getErrorLineCount());
         assertEquals(0, result.getSkippedLineCount());
         assertEquals(336, result.getSuccessLineCount());
         assertEquals(336, result.getTotalLineCount());
     }
+
+    @Test
+    public void testImportIgnoreSurroundingSpaces() throws OperationException, InterruptedException {
+        Map<String, Object> params = Map.of("path", testFolder.getPathAsString());
+
+        chain = new OperationChain("test-chain");
+        chain.add(CSVImportOperation.ID).from(params);
+
+        CSVImportResult result;
+        File csv = FileUtils.getResourceFileFromContext(DOCS_SURROUNDING_SPACES_CSV);
+        Blob blob = new FileBlob(csv);
+        String importId;
+        try (OperationContext ctx = new OperationContext(session)) {
+            ctx.setInput(blob);
+            importId = (String) service.run(ctx, chain);
+
+        }
+
+        assertNotNull(importId);
+
+        boolean completed = false;
+        do {
+            chain = new OperationChain("test-chain");
+            chain.add(CSVImportStatusOperation.ID);
+            CSVImportStatus status;
+            try (OperationContext ctx = new OperationContext(session)) {
+                ctx.setInput(importId);
+                status = (CSVImportStatus) service.run(ctx, chain);
+            }
+
+            assertNotNull(status);
+            completed = status.isComplete();
+            if (!completed) {
+                Thread.sleep(100);
+            }
+        } while (!completed);
```

Comment:
You removed the timeout mechanism present in other tests which makes your test dangerous for the CI if operation never completes. In such cases, we should use [awaitility](https://github.com/awaitility/awaitility) framework which handles correctly thread and wait (instead of `Thread#sleep`).

On the other hand, we could put this test in the service test class 🤔.

Reply:
Yes, if using the operation, the timeout should be kept.
But indeed, testing with the service does make more sense here.
I'll check with @NourNuxeo.

Reply:
I thought it was specific for the big import that was supposed to be fast enough.
So are all tests supposed to have a timeout mechanism ?

Reply:
Well, even if your CSV is small, you have no guarantee that the import will end at some point.
So you might be trapped in an infinite loop...
This is specific to the fact that you are using this pattern:
```
do {
    Thread.sleep(1000)
} while (condition)
```
The condition might never be verified....

Reply:
Agreed, it would be better to make sure we stop and fail after a reasonable timeout (that should never happen but you never know) like 10s.